syntax = "proto3";

option go_package = "services/proto";

import "services/proto/general.proto";

message UpdateDetails {
  int64 user_id = 1;
  int64 article_id = 2;
  string body = 3;
  string title = 4;
  repeated string tags = 5;
  string summary = 6;
}

message ReceivedUpdateDetails {
  string ap_id = 1;
  string body = 2;
  string title = 3;
  string summary = 4;
}

// Service for sending and receiving server-to-server update activities.
service S2SUpdate {
  rpc ReceiveUpdateActivity(ReceivedUpdateDetails) returns (GeneralResponse);
  rpc SendUpdateActivity(UpdateDetails) returns (GeneralResponse);
}
